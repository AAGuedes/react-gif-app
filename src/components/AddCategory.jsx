import { useState } from 'react';


export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const value = inputValue.trim();
        if (value.length <= 1) return;

        setInputValue('');
        onNewCategory(value);
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                className='form-control mb-3'
                type="text"
                placeholder="Buscar gifs..."
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
