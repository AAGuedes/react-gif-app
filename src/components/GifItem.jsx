

export const GifItem = ({ title, url }) => {

    return (
        <>
            <div className="card p-0 col-3 mx-2 my-2">
                <img className="card-img-top" src={url} alt={title} />
                <div className="card-body">
                    <p className="card-text text-center">{title}</p>
                </div>
            </div>
        </>
    )
}
