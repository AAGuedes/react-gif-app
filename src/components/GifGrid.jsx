import { useEffect, useState } from 'react';

import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';


export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(category);
        setImages(newImages);
    }

    useEffect(() => {
        getImages();
    }, []);

    return (
        <>
            <h3 className='text-center mt-5 mb-3'>{category}</h3>
            <div className='d-flex row d-flex justify-content-evenly'>
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}
