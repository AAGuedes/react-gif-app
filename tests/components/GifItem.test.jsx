import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas de <GifItem />', () => {

    const title = 'Title';
    const url = 'https://www.localhost.es/';

    test('Debe coincidir con la snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

});
