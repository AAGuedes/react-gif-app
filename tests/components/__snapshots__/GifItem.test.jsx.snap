// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem /> Debe coincidir con la snapshot 1`] = `
<div>
  <div
    class="card p-0 col-3 mx-2 my-2"
  >
    <img
      alt="Title"
      class="card-img-top"
      src="https://www.localhost.es/"
    />
    <div
      class="card-body"
    >
      <p
        class="card-text text-center"
      >
        Title
      </p>
    </div>
  </div>
</div>
`;
